### ISSUE TITLE
git deps: use branch "main" if "master" does not exist

### ISSUE THREAD

**Issue by birkenfeld** [rust-lang/cargo#8946]
With more projects switching to their primary branch being called "main" instead of "master", simply adding a git dependency will fail.  This is a usability issue, and since I assume the Rust project supports the use of this alternate nomenclature, it should do its part to support it.

E.g. with `ureq`:

```
error: failed to get `ureq` as a dependency of package ...

Caused by:
  failed to load source for dependency `ureq`

Caused by:
  Unable to update https://github.com/algesten/ureq

Caused by:
  failed to find branch `master`

Caused by:
  cannot locate remote-tracking branch 'origin/master'; class=Reference (4); code=NotFound (-3)
```

**Comment by ehuss** [comment]
Thanks for the report.  This is loosely tracked in issue #3517.  Support as already been implemented in #8522, but this requires a change in the format of `Cargo.lock`, so it takes a long time to roll out the transition.  Support for it will probably start phasing in sometime next year.

**Comment by birkenfeld** [comment]
Thanks, I'll close this then.
